django/middleware/csrf.py

```diff
@@ -437,15 +434,15 @@ def process_view(self, request, callback, callback_args, callback_kwargs):
         return self._accept(request)
 
     def process_response(self, request, response):
-        if not getattr(request, 'csrf_cookie_needs_reset', False):
-            if getattr(response, 'csrf_cookie_set', False):
-                return response
-
-        if not request.META.get("CSRF_COOKIE_USED", False):
+        # Prevent _set_token() from being called twice.
```

Comment:
`_set_token()` is only called here (below). This is within `process_response()`. The only reasonable case where it would be called twice is when both middleware and decorator are used (the less reasonable case is when the decorator is somehow applied more than once to the same view). I'd add to the comment something like 
```suggestion
        # Prevent _set_token() from being called twice (e.g. when both decorator and middleware used)
```

Reply:
Thanks, @shaib. I updated the comment with your suggestion.